lms/static/js/edxnotes/plugins/accessibility.js

```diff
@@ -0,0 +1,247 @@
+            $(field).attr({
+                'tabindex': -1,
+                'id': ariaNoteId,
+                'role': 'note',
+                'class': 'annotator-note'
+            });
+        },
+
+        focusOnHighlightedText: function (event) {
+            var id = this.annotator.element.find('.annotator-viewer')
+                                           .find('.annotator-note')
+                                           .attr('id');
+            $('.annotator-hl[aria-describedby=' + id + ']').focus();
+            event.preventDefault();
+        },
+
+        getViewerTabControls: function () {
+            var viewer, notes, viewerControls, editButtons, delButtons, closeButtons, tabControls = [], i;
+
+            // Viewer elements
+            viewer = this.annotator.element.find('.annotator-viewer');
+            notes = viewer.find('.annotator-note');
+            viewerControls = viewer.find('.annotator-controls');
+            editButtons = viewerControls.find('.annotator-edit');
+            delButtons = viewerControls.find('.annotator-delete');
+            closeButtons = viewerControls.find('.annotator-close');
+
+            // Note and ddit, delete and close buttons always come in quadruplets
+            for (i = 0; i < editButtons.length; i++) {
+                tabControls.push($(notes.get(i)));
+                tabControls.push($(editButtons.get(i)));
+                tabControls.push($(delButtons.get(i)));
+                tabControls.push($(closeButtons.get(i)));
+            }
+
+            return tabControls;
+        },
+
+        getEditorTabControls: function () {
+            var editor, editorControls, textArea, save, cancel, tabControls = [];
+
+            // Editor elements
+            editor = this.annotator.element.find('.annotator-editor');
+            editorControls = editor.find('.annotator-controls');
+            textArea = editor.find('.annotator-listing')
+                             .find('.annotator-item')
+                             .first()
+                             .children('textarea');
+            save  = editorControls.find('.annotator-save');
+            cancel = editorControls.find('.annotator-cancel');
+
+            tabControls.push($(textArea.get(0)), $(save.get(0)), $(cancel.get(0)));
+
+            return tabControls;
+        },
+
+        focusOnNextTabControl: function (tabControls, tabControl) {
+            var nextIndex;
+
+            _.each(tabControls, function (element, index) {
+                if (element.is(tabControl)) {
+                    nextIndex = index === tabControls.length - 1 ? 0 : index + 1;
+                    tabControls[nextIndex].focus();
+                }
+
+            });
+        },
+
+        focusOnPreviousTabControl: function (tabControls, tabControl) {
+            var previousIndex;
+            _.each(tabControls, function (element, index) {
+                if (element.is(tabControl)) {
```

Comment:
@polesye 
> is it necessary to do this check?

This check is used only to get the index of the element that has current focus in tabControls array. I'll try to find a better way, I also do not like that _.each statement here.